From b6bb46ef161e1c3890f517fdb04f081926fc1a08 Mon Sep 17 00:00:00 2001
From: Seevalamuthu Mariappan <seevalam@codeaurora.org>
Date: Tue, 23 Mar 2021 19:57:43 +0530
Subject: [PATCH] hostapd: update missing 5.9GHz channels

Update channel 165 and 173 in allowed channel.
This fixes below upstream commit,
833cdbe - "Add support for new 5 GHz channels 173 and 177"

Signed-off-by: Seevalamuthu Mariappan <seevalam@codeaurora.org>
---
 src/common/hw_features_common.c | 2 +-
 wpa_supplicant/wpa_supplicant.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: hostapd-2021-12-13-b26f5c0f/wpa_supplicant/wpa_supplicant.c
===================================================================
--- hostapd-2021-12-13-b26f5c0f.orig/wpa_supplicant/wpa_supplicant.c
+++ hostapd-2021-12-13-b26f5c0f/wpa_supplicant/wpa_supplicant.c
@@ -2540,8 +2540,8 @@ void ibss_mesh_setup_freq(struct wpa_sup
 	enum hostapd_hw_mode hw_mode;
 	struct hostapd_hw_modes *mode = NULL;
 	int ht40plus[] = { 1, 2, 3, 4, 5, 6, 36, 44, 52, 60, 100, 108, 116, 124, 132, 149, 157,
-			   184, 192 };
-	int bw80[] = { 5180, 5260, 5500, 5580, 5660, 5745, 5955,
+			   165, 173, 184, 192 };
+	int bw80[] = { 5180, 5260, 5500, 5580, 5660, 5745, 5825, 5955,
 		       6035, 6115, 6195, 6275, 6355, 6435, 6515,
 		       6595, 6675, 6755, 6835, 6915, 6995 };
 	int bw160[] = { 5955, 6115, 6275, 6435, 6595, 6755, 6915 };
Index: hostapd-2021-12-13-b26f5c0f/src/ap/acs.c
===================================================================
--- hostapd-2021-12-13-b26f5c0f.orig/src/ap/acs.c
+++ hostapd-2021-12-13-b26f5c0f/src/ap/acs.c
@@ -369,7 +369,7 @@ acs_survey_chan_interference_factor(stru
 static int acs_usable_bw40_chan(const struct hostapd_channel_data *chan)
 {
 	const int allowed[] = { 5180, 5220, 5260, 5300, 5500, 5540, 5580, 5620,
-				5660, 5745, 5785, 4920, 4960, 5955, 5995, 6035,
+				5660, 5745, 5785, 5825, 5865, 4920, 4960, 5955, 5995, 6035,
 				6075, 6115, 6155, 6195, 6235, 6275, 6315, 6355,
 				6395, 6435, 6475, 6515, 6555, 6595, 6635, 6675,
 				6715, 6755, 6795, 6835, 6875, 6915, 6955, 6995,
@@ -386,7 +386,7 @@ static int acs_usable_bw40_chan(const st
 
 static int acs_usable_bw80_chan(const struct hostapd_channel_data *chan)
 {
-	const int allowed[] = { 5180, 5260, 5500, 5580, 5660, 5745, 5955, 6035,
+	const int allowed[] = { 5180, 5260, 5500, 5580, 5660, 5745, 5825, 5955, 6035,
 				6115, 6195, 6275, 6355, 6435, 6515, 6595, 6675,
 				6755, 6835, 6915, 6995 };
 	unsigned int i;
@@ -401,7 +401,7 @@ static int acs_usable_bw80_chan(const st
 
 static int acs_usable_bw160_chan(const struct hostapd_channel_data *chan)
 {
-	const int allowed[] = { 5180, 5500, 5955, 6115, 6275, 6435, 6595, 6755,
+	const int allowed[] = { 5180, 5500, 5745, 5955, 6115, 6275, 6435, 6595, 6755,
 				6915 };
 	unsigned int i;
 
